sdk/core/Azure.Core.Experimental/src/DynamicJsonOptions.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core.Dynamic
+{
+    /// <summary>
+    /// Provides the ability for the user to define custom behavior when accessing JSON through a dynamic layer.
+    /// </summary>
+    public struct DynamicJsonOptions
+    {
+        /// <summary>
+        /// Creates a new instance of DynamicJsonOptions.
+        /// </summary>
+        public DynamicJsonOptions() { }
+
+        /// <summary>
+        /// Specifies how properties on <see cref="DynamicJson"/> will get accessed in the underlying JSON buffer.
+        /// </summary>
+        public DynamicJsonPropertyCasing PropertyCasing { get; set; } = DynamicJsonPropertyCasing.Default;
```

Comment:
Are we going to have more options? If not, should the methods just take the enum as opposed to the options? If yes, then maybe we can have overlaods taking just the enum, or a cast from enum to options?

Reply:
I can imagine more options we might want - e.g. options to the JsonSerializer we use to do type conversions, possibly options to use a different delimiter than `.` if needed by an API, "StandardFormat" options, etc.  But an options type could be added later if we think casing is the most common thing people would want to configure, and taken as an overload (or provide a cast) per your suggestion.

Reply:
I kept the options type in this version because of the other options I think we'll want to add, but tell me if you hate it or feel strongly about the cast.  I'm leaning against the cast because I don't know if this would be the most common setting people would want to change from the options, and I didn't see casts like this on BCL types.